gulpTasks/doc.js

```diff
@@ -24,40 +32,65 @@ gulp.task('doc', ['copyBinToDoc', 'buildPlayground'], function() {
   const src = app.expandInputFiles(['src']);
   const project = app.convert(src);
   app.generateDocs(project, 'docgen');
-  app.generateJson(project, './bin/docgen/docgen.json', 'https://coveo.github.io/search-ui/');
+  app.generateJson(project, docgenJsonPath, 'https://coveo.github.io/search-ui/');
   return gulp.src('./readme.png').pipe(gulp.dest('./docgen'));
 });
 
 gulp.task('cleanGeneratedThemesFiles', () => {
   return del(['./docs/theme/assets/gen/**/*']);
 });
 
-gulp.task('copyBinToDoc', ['cleanGeneratedThemesFiles'], function() {
+gulp.task('copyBinToDoc', ['cleanGeneratedThemesFiles'], () => {
   return gulp.src('./bin/{js,image,css}/**/*').pipe(gulp.dest('./docs/theme/assets/gen'));
 });
 
 gulp.task('buildPlayground', shell.task(['node node_modules/webpack/bin/webpack.js --config ./webpack.playground.config.js']));
 
-function copyFile(source, target, cb) {
-  var cbCalled = false;
+gulp.task('testDoc', () => {
+  const docgenJson = JSON.parse(fs.readFileSync(docgenJsonPath));
+  if (!docgenJson || !docgenJson.length) {
+    throw new Error('Invalid object');
+  }
 
-  var rd = fs.createReadStream(source);
-  rd.on('error', function(err) {
-    done(err);
-  });
-  var wr = fs.createWriteStream(target);
-  wr.on('error', function(err) {
-    done(err);
-  });
-  wr.on('close', function(ex) {
-    done();
-  });
-  rd.pipe(wr);
+  checkAllAttributesAreDefinedCorrectly(docgenJson);
+  checkSpecificKeywords(docgenJson);
+  checkSpecificElements(docgenJson);
+});
 
-  function done(err) {
-    if (!cbCalled) {
-      cb(err);
-      cbCalled = true;
+function checkAllAttributesAreDefinedCorrectly(docgenJson) {
+  const attributes = [
+    { name: 'name', types: ['string'] },
+    { name: 'notSupportedIn', types: ['string', 'object'] },
+    { name: 'comment', types: ['string'] },
+    { name: 'type', types: ['string'] },
+    { name: 'constrainedValues', types: ['object'] },
+    { name: 'miscAttributes', types: ['object'] }
+  ];
+
+  _.each(docgenJson, element =>
+    _.each(attributes, ({ name, types }) => {
+      if (!_.has(element, name) || !_.contains(types, typeof element[name])) {
+        throw new Error(`Invalid or missing attribute "${name}" for doc element "${JSON.stringify(element)}"`);
+      }
+    })
+  );
+}
+
+function checkSpecificKeywords(docgenJson) {
+  const keywords = ['Analytics', 'QueryBuilder', 'Recommendation'];
+  _.each(keywords, keyword => {
+    if (!_.find(docgenJson, doc => doc.name.includes(keyword))) {
```

Comment:
Small point, but perhaps a more precise check would be to using `===` instead of `includes`. Although unlikely, I am thinking of the case where say we have `AdvancedAnalytics` present but `Analytics` is missing. In that situation the test would still pass. Also, `_.findWhere` could be another approach. e.g.

```
const keywordFound = _.findWhere(docgenJson, { name: keyword } );

if (!keywordFound) {
    throw new Error()
}
```

Reply:
I agree that `findWhere` is a better approach if `===` is used to find the exact component. The goal here as @olamothe mentioned was a partial match to find a couple keywords to get the general idea that the content makes sense, not to find an exact component.

Reply:
I could have juste done also quick a search in the json to find those keywords. Maybe this solution is unclear.